### ISSUE TITLE
[icu] Build error

### ISSUE THREAD

**Issue by acgetchell** [microsoft/vcpkg#30022]
**Host Environment**

- OS: macOS 13.2.1
- Compiler: revision
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin22.3.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

**To Reproduce**

`./vcpkg install icu`

**Failure logs**

- Cut and paste the appropriate build messages from the console output.
- Attach ***every failure log file*** mentioned in the console output.

Installing 5/149 icu:x64-osx...
Building icu[core,tools]:x64-osx...
-- Using cached icu4c-72_1-src.tgz.
-- Cleaning sources at /Users/adam/projects/vcpkg/buildtrees/icu/src/c-72_1-src-deba61ed52.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/adam/projects/vcpkg/downloads/icu4c-72_1-src.tgz
-- Applying patch disable-escapestr-tool.patch
-- Applying patch remove-MD-from-configure.patch
-- Applying patch fix_parallel_build_on_windows.patch
-- Applying patch fix-extra.patch
-- Applying patch mingw-dll-install.patch
-- Applying patch disable-static-prefix.patch
-- Applying patch fix-win-build.patch
-- Using source at /Users/adam/projects/vcpkg/buildtrees/icu/src/c-72_1-src-deba61ed52.clean
-- Found external ninja('1.11.1').
-- Getting CMake variables for x64-osx-dbg
-- Getting CMake variables for x64-osx-rel
-- Generating configure for x64-osx
-- Finished generating configure for x64-osx
-- Configuring x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /bin/bash -c "V=1 ./../src/c-72_1-src-deba61ed52.clean/source/configure  \"--disable-samples\" \"--disable-tests\" \"--disable-layoutex\" \"--enable-tools\" \"--disable-silent-rules\" \"--verbose\" \"--disable-shared\" \"--enable-static\" \"--enable-debug\" \"--disable-release\" \"--prefix=/Users/adam/projects/CDT-plusplus/vcpkg_installed/x64-osx/debug\" \"--bindir=\\${prefix}/../tools/icu/debug/bin\" \"--sbindir=\\${prefix}/../tools/icu/debug/sbin\" \"--libdir=\\${prefix}/lib\" \"--includedir=\\${prefix}/../include\" \"--datarootdir=\\${prefix}/share/icu\""
    Working Directory: /Users/adam/projects/vcpkg/buildtrees/icu/x64-osx-dbg
    Error code: 2
    See logs for more information:
      /Users/adam/projects/vcpkg/buildtrees/icu/config-x64-osx-dbg-config.log
      /Users/adam/projects/vcpkg/buildtrees/icu/config-x64-osx-dbg-out.log
      /Users/adam/projects/vcpkg/buildtrees/icu/config-x64-osx-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:812 (vcpkg_execute_required_process)
  ports/icu/portfile.cmake:55 (vcpkg_configure_make)
  scripts/ports.cmake:147 (include)

error: building icu:x64-osx failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+icu
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[icu]+Build+error
Include '[icu] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2023-02-16-12e657924d99511514c0287ca5ce46882d3657c7
    vcpkg-scripts version: 8daf70c56 2023-03-03 (23 hours ago)

**Additional context**

Add any other context about the problem here, such as what you have already tried to resolve the issue.

[config-x64-osx-dbg-config.log](https://github.com/microsoft/vcpkg/files/10889509/config-x64-osx-dbg-config.log)

[config-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/10889510/config-x64-osx-dbg-out.log)

[config-x64-osx-dbg-err.log](https://github.com/microsoft/vcpkg/files/10889511/config-x64-osx-dbg-err.log)

**Comment by dg0yt** [comment]
The error message looks unusual:
~~~
./../src/c-72_1-src-deba61ed52.clean/source/configure: line 4548: syntax error near unexpected token `0.20'
./../src/c-72_1-src-deba61ed52.clean/source/configure: line 4548: `PKG_PROG_PKG_CONFIG(0.20)'
~~~
However, your `PATH` indicates that you use spack and probably brew. I would suggest you retry with a reduced PATH. Only use brew in addition to Apple's toolchain and system paths - this a more usual environment.

**Comment by TrueBrain** [comment]
I am seeing the exact same issue when building on Linux with `quay.io/pypa/manylinux2014_x86_64` build environment. Looking inside `configure`, it looks like this:

```
(..)
ac_compiler_gnu=$ac_cv_c_compiler_gnu


# pkg-config is needed for harfbuzz support
PKG_PROG_PKG_CONFIG(0.20)
PKG_CHECK_MODULES(ICULEHB, icu-le-hb, have_icu_le_hb=true, :)

# Ensure that if CXXFLAGS/CFLAGS were not set when calling configure, set it correctly based on (enable/disable) debug or release option
# The release mode use is the default one for autoconf
if test "$GCC" = yes; then
(..)
```

It almost feels like a bit of CMake syntax got injected in a bash (`configure` file) script. Rather unusual. Will see if I can look a bit deeper what is actually going on.

**Comment by dg0yt** [comment]
Interesting... Does it pick up another pkg.m4 perhaps? I can't find one in vcpkg CI file list for x64-linux. But some comments/logs showed additional package managers.

**Comment by TrueBrain** [comment]
Okay, some more poking around, it seems if you have `/usr/share/aclocal` and `/usr/local/share/aclocal`, the tool `aclocal` only looks in the latter, not in the first. So what solved it for me was the silly:

`cp /usr/share/aclocal/* /usr/local/share/aclocal/`

So I am not totally sure if this is a vcpkg issue, or an issue caused by vcpkg, or just a silly thing about `aclocal.

PS: this is on a completely clean docker container based on `quay.io/pypa/manylinux2014_x86_64`, which is basically CentOS 7.

**Comment by TrueBrain** [comment]
Yeah, okay, this really is `aclocal` and the default installation here. Both `/usr/share/aclocal` and `/usr/local/share/aclocal` exist, and `aclocal` really only looks in the latter.

I don't know anything about the autotools family, other than the experiments I just did. So I will just use my dirty `cp ..` solution for the CI jobs that need this .. no clue what a proper solution would look like :)

In short:
- Make sure you have `pkgconf` installed
- Make sure that `pkg.m4` is in a folder `aclocal` can find, which seems to be `/usr/local/share/aclocal` if it exists, or `/usr/share/aclocal` otherwise.
